be/src/formats/parquet/column_reader.cpp

```diff
@@ -123,22 +132,259 @@ class ScalarColumnReader : public ColumnReader {
         _reader->get_levels(def_levels, rep_levels, num_levels);
     }
 
-    Status get_dict_values(Column* column) override { return _reader->get_dict_values(column); }
-
     Status get_dict_values(const std::vector<int32_t>& dict_codes, const NullableColumn& nulls,
                            Column* column) override {
         return _reader->get_dict_values(dict_codes, nulls, column);
     }
 
     void set_need_parse_levels(bool need_parse_levels) override { _reader->set_need_parse_levels(need_parse_levels); }
 
+    bool try_to_use_dict_filter(ExprContext* ctx, bool is_output_column, const SlotId slotId,
+                                const std::vector<std::string>& sub_field_path, const size_t& layer) override {
```

Comment:
what does `layer` mean?

Reply:
for sutrct column there may be many layers, such as a.b.c.d, so the layer is the layer in the path from the root to the leaf